pkg/vpcmodel/grouping_test.go

```diff
@@ -390,40 +390,3 @@ func TestConfigSelfLoopCliqueLace(t *testing.T) {
 	fmt.Println(groupingStr)
 	fmt.Println("done")
 }
-func configSubnetSelfLoop() (*VPCConfig, *VPCsubnetConnectivity) {
-	res := &VPCConfig{Nodes: []Node{}}
-	res.Nodes = append(res.Nodes,
-		&mockNetIntf{cidr: "10.0.20.5/32", name: "vsi1"},
-		&mockNetIntf{cidr: "10.3.20.6/32", name: "vsi2"},
-		&mockNetIntf{cidr: "10.7.20.7/32", name: "vsi3"})
-
-	res.NodeSets = append(res.NodeSets, &mockSubnet{"10.0.20.0/22", "subnet1", []Node{res.Nodes[0]}},
-		&mockSubnet{"10.3.20.0/22", "subnet2", []Node{res.Nodes[1]}},
-		&mockSubnet{"10.7.20.0/22", "subnet3", []Node{res.Nodes[2]}})
-
-	res1 := &VPCsubnetConnectivity{AllowedConnsCombined: GeneralConnectivityMap{}}
-	res1.AllowedConnsCombined.updateAllowedConnsMap(res.NodeSets[0], res.NodeSets[1], common.NewConnectionSet(true))
-	res1.AllowedConnsCombined.updateAllowedConnsMap(res.NodeSets[0], res.NodeSets[2], common.NewConnectionSet(true))
-	res1.AllowedConnsCombined.updateAllowedConnsMap(res.NodeSets[1], res.NodeSets[0], common.NewConnectionSet(true))
-	res1.AllowedConnsCombined.updateAllowedConnsMap(res.NodeSets[1], res.NodeSets[2], common.NewConnectionSet(true))
-	res1.AllowedConnsCombined.updateAllowedConnsMap(res.NodeSets[2], res.NodeSets[0], common.NewConnectionSet(true))
-	res1.AllowedConnsCombined.updateAllowedConnsMap(res.NodeSets[2], res.NodeSets[1], common.NewConnectionSet(true))
-
-	return res, res1
-}
-
-func TestSubnetSelfLoop(t *testing.T) {
```

Comment:
why is this test removed?

Reply:
because now its too much trouble keeping it - the subnet needs an attached VPC with an UID - and we have this test, of clique subnets, in the testing files

Reply:
now that we have abstract type for VPC, no need for another mock, we can add the VPC and its uid for the subnets in the test.. I think there is value for covering this functionality in unit tests, and not only rely on end to end tests..

Reply:
Did you add any unit test for the abstract `VPC()` and `Subnet()` I can use as reference?

Reply:
no, but I think it should be straight forward

Reply:
Not sure what you had in mind, but to my understanding we still need to implement VPCResourceIntf outside ibmvpc; which still requires a mock. I've added mockVPCIntf and restored the test